staging/src/k8s.io/api/rbac/v1/types.go

```diff
@@ -57,7 +57,10 @@ type PolicyRule struct {
 	// Resources is a list of resources this rule applies to.  ResourceAll represents all resources.
 	// +optional
 	Resources []string `json:"resources,omitempty" protobuf:"bytes,3,rep,name=resources"`
-	// ResourceNames is an optional white list of names that the rule applies to.  An empty set means that everything is allowed.
+	// ResourceNames is an optional white list of object names that the rule applies to.  An empty set means that everything is allowed.
+	// The object name is taken from the request URL's path if present
+	// there, otherwise from the relevant field selectoer in the URL's
+	// query if present there, otherwise is the empty string.
```

Comment:
this also seems out of place... the RBAC types do not describe how all attributes are determined (verbs, API groups, resources, etc)... that is the responsibility of whatever is calling the authorization system (either the subjectaccessreview API caller, or the API server)

Reply:
I actually think this is a relevant thing for consumers / producers of these api objects to want to know. I'd just preface it with "If the request attributes to be matched are sourced from an API request, then..."